### ISSUE TITLE
Interfaces don't work for Pydantic types

### ISSUE THREAD

**Issue by paul-sud** [strawberry-graphql/strawberry#1244]
If you try to use interfaces with types converted from Pydantic:

```python
from typing import List

import strawberry
from pydantic import BaseModel


@strawberry.interface
class Cheese:
    name: str

class SwissModel(BaseModel):
    canton: str

@strawberry.experimental.pydantic.type(model=SwissModel, fields=["canton"])
class Swiss(Cheese):
    pass

class ItalianModel(BaseModel):
    province: str

@strawberry.experimental.pydantic.type(model=ItalianModel, fields=["province"])
class Italian(Cheese):
    pass

@strawberry.type
class Root:
    @strawberry.field
    def assortment(self) -> List[Cheese]:
        return [
            Italian(name="Asiago", province="Friuli"),
            Swiss(name="Tomme", canton="Vaud"),
        ]

schema = strawberry.Schema(query=Root, types=[Swiss, Italian])

query = """{
    assortment {
        name
        ... on Italian { province }
        ... on Swiss { canton }
    }
}"""

result = schema.execute_sync(query)

assert not result.errors
assert result.data["assortment"] == [
    {"name": "Asiago", "province": "Friuli"},
    {"canton": "Vaud", "name": "Tomme"},
]
```

you get the following error:

```
Fragment cannot be spread here as objects of type 'Cheese' can never be of type 'Italian'.

GraphQL request:4:9
3 |         name
4 |         ... on Italian { province }
  |         ^
5 |         ... on Swiss { canton }
Stack (most recent call last):
  File "/Users/paul/gitclones/snoberry/fail.py", line 44, in <module>
    result = schema.execute_sync(query)
Fragment cannot be spread here as objects of type 'Cheese' can never be of type 'Swiss'.

GraphQL request:5:9
4 |         ... on Italian { province }
5 |         ... on Swiss { canton }
  |         ^
6 |     }
Stack (most recent call last):
  File "/Users/paul/gitclones/snoberry/fail.py", line 44, in <module>
    result = schema.execute_sync(query)
Traceback (most recent call last):
  File "/Users/paul/gitclones/snoberry/fail.py", line 46, in <module>
    assert not result.errors
AssertionError
```

This example is adapted from https://github.com/strawberry-graphql/strawberry/blob/5dbbb04c8542526387d353fb871307ce61bd6546/tests/schema/test_interface.py#L9

I believe the issue is that the bases of `Italian` and `Swiss`, in this case `Cheese`, are not propagated to the dataclass during when they are wrapped: https://github.com/strawberry-graphql/strawberry/blob/5dbbb04c8542526387d353fb871307ce61bd6546/strawberry/experimental/pydantic/object_type.py#L132

**Comment by paul-sud** [comment]
PR: https://github.com/strawberry-graphql/strawberry/pull/1245

**Comment by paul-sud** [comment]
This issue is being handled in https://github.com/strawberry-graphql/strawberry/pull/1241

**Comment by paul-sud** [comment]
Fixed in 0.77.9, closing: https://github.com/strawberry-graphql/strawberry/releases/tag/0.77.9
